src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultIDictionaryConverter.cs

```diff
@@ -8,7 +8,7 @@ namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class DefaultIDictionaryConverter : JsonDictionaryConverter
     {
-        public override IDictionary CreateFromDictionary(ref ReadStack state, IDictionary sourceDictionary, JsonSerializerOptions options)
+        public override object CreateFromDictionary(ref ReadStack state, IDictionary sourceDictionary, JsonSerializerOptions options)
```

Comment:
Why the change to `object`? Don't all dictionaries implement `IDictionary` (non generic)?

Reply:
Turns out that [`IDictionary<TKey, TValue`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2?view=netcore-3.0) does not. So we can't support (user-defined) types that implement only  `IDictionary<TKey, TValue>` with this converter mechanism.

I think it's better to change to object here than to implement a new converter mechanism for just these types, unless boxing to `object` is significantly worse than to `IDictionary` or `IDictionary<,>`.

In the meantime, I've added a comment to `JsonDictionaryConverter` stating that it's usage is generalized to deserializing (some) collections that store key-value pairs (not including `KeyValuePair<,>`) -- if it's okay to leave as object.

Reply:
OK makes sense